File: content/problems/3099-harshad-number/solutions/digit-sum-and-divisibility-check/Solution.java
class Solution {
    public int sumOfTheDigitsOfHarshadNumber(int x){
        int original=x,digitSum=0;
        while(x>0){
            digitSum+=x%10;
            x/=10;
        }
        return original%digitSum==0?digitSum:-1;
    }
}

File: content/problems/3099-harshad-number/solutions/digit-sum-and-divisibility-check/Solution.kt
class Solution {
    fun sumOfTheDigitsOfHarshadNumber(x:Int):Int{
        var value=x;
        var digitSum=0;
        while(value>0){
            digitSum+=value%10;
            value/=10
        };
        return if(x%digitSum==0)digitSum else -1
    }
}
